suricata/update/commands/checkversions.py

```diff
@@ -0,0 +1,57 @@
+# Copyright (C) 2017 Open Information Security Foundation
+#
+# You can copy, redistribute or modify this Program under the terms of
+# the GNU General Public License version 2 as published by the Free
+# Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# version 2 along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.
+
+import os.path
+import logging
+from suricata.update import sources, engine
+
+logger = logging.getLogger()
+
+
+def register(parser):
+    parser.set_defaults(func=check_version)
+
+
+def check_version(suricata_version):
+    index_filename = sources.get_index_filename()
+    if not os.path.exists(index_filename):
+        logger.warning("No index exists, will use bundled index.")
+        logger.warning("Please run suricata-update update-sources.")
+    index = sources.Index(index_filename)
+    version = index.get_versions()
+    recommended = engine.parse_version(version['suricata']['recommended'])
+    if suricata_version.full in recommended.full:
```

Comment:
is this an exact match? or would '4.1.4' in '4.1.4.1' also match? We don't use the latter notation at this time, but if we can have the logic be robust against such changes I think it would be great.

Reply:
This uses the existing version parser where `.full` is the just the raw version string passed in, provided it parsed OK.  Right now this only supports 4.1.4, not 4.1.4.1.  I'd suggest doing that as a separate task.

Reply:
Not sure I follow. An exact string match would accomplish both, right?

Reply:
It would also make sure '3.1' in '4.3.1' wouldn't match.

Reply:
> is this an exact match? or would '4.1.4' in '4.1.4.1' also match? We don't use the latter notation at this time, but if we can have the logic be robust against such changes I think it would be great.

It's not the exact match. Here '4.1.4' in '4.1.4.1' would match if we use '4.1.4.1' as recommended version. I think we should have an exact string match here.